librz/core/rop.c

```diff
@@ -302,11 +302,11 @@ RZ_API RZ_OWN RzRopGadgetInfo *rz_core_rop_gadget_info_new(const ut64 address) {
  *
  * Frees the memory allocated for an RzRopGadgetInfo object, including its modified registers and dependencies.
  */
-RZ_API void rz_core_rop_gadget_info_free(RZ_NULLABLE RzRopGadgetInfo *gadget_info) {
+RZ_API void rz_core_rop_gadget_info_free(RZ_NONNULL RzRopGadgetInfo *gadget_info) {
```

Comment:
I recommend allowing NULL in the argument and just do a quick return if it is. Also remove the assert then. Any `_free()` function should work with NULL just fine.